.travis.yml

```diff
@@ -4,37 +4,46 @@ language: generic
 sudo: required
 
 env:
-  DOCKER_COMPOSE_VERSION: 1.16.1
+  global:
+    - DOCKER_COMPOSE_VERSION=1.26.2
+    - ANSIBLE_VERSION=2.9.9
+  jobs:
+    - PLATFORM=centos PLAYBOOK=user.yml DOCKER_CACHE_DIR=$HOME/docker/$PLATFORM
+    - PLATFORM=debian PLAYBOOK=user.yml DOCKER_CACHE_DIR=$HOME/docker/$PLATFORM
+    - PLATFORM=ubuntu PLAYBOOK=user.yml DOCKER_CACHE_DIR=$HOME/docker/$PLATFORM
+    - PLATFORM=centos PLAYBOOK=root.yml DOCKER_CACHE_DIR=$HOME/docker/$PLATFORM
+    - PLATFORM=debian PLAYBOOK=root.yml DOCKER_CACHE_DIR=$HOME/docker/$PLATFORM
+    - PLATFORM=ubuntu PLAYBOOK=root.yml DOCKER_CACHE_DIR=$HOME/docker/$PLATFORM
 
 services:
   - docker
 
 cache:
   pip: true
   directories:
-    - $HOME/docker
+    - $DOCKER_CACHE_DIR
 
 before_install:
-  - if [[ -d $HOME/docker ]]; then ls $HOME/docker/*.tar.gz | xargs -I {file} sh -c "zcat {file} | docker load"; fi
+  - if [[ -d $DOCKER_CACHE_DIR ]]; then ls $DOCKER_CACHE_DIR/*.tar.gz | xargs -I {file} sh -c "zcat {file} | docker load"; fi
   - sudo rm /usr/local/bin/docker-compose
   - curl -L https://github.com/docker/compose/releases/download/$DOCKER_COMPOSE_VERSION/docker-compose-`uname -s`-`uname -m` > docker-compose
   - chmod +x docker-compose
   - sudo mv docker-compose /usr/local/bin
 
 install:
-  - sudo pip install ansible==2.4.6.0 > /dev/null 2>&1
+  - sudo pip install ansible==$ANSIBLE_VERSION > /dev/null 2>&1
 
 script:
   - cd tests
-  - docker-compose up -d
-  - ansible-playbook user.yml -l centos
-  - ansible-playbook user.yml -l debian
-  - ansible-playbook user.yml -l ubuntu
+  - docker-compose config --services | grep $PLATFORM | xargs docker-compose up -d
+  - ansible-playbook $PLAYBOOK -l $PLATFORM
+  - docker-compose down
 
 before_cache:
   - >
-    mkdir -p $HOME/docker && docker images -a --filter='dangling=false' --format '{{.Repository}}:{{.Tag}} {{.ID}}'
-    | xargs -n 2 -t sh -c 'test -e $HOME/docker/$1.tar.gz || docker save $0 | gzip -2 > $HOME/docker/$1.tar.gz'
+    mkdir -p $DOCKER_CACHE_DIR && docker images -a --filter='dangling=false' --format '{{.Repository}}:{{.Tag}} {{.ID}}' "tests_$PLATFORM*"
+    | xargs -n 2 -t sh -c 'test -e $DOCKER_CACHE_DIR/$1.tar.gz || docker save $0 | gzip -2 > $DOCKER_CACHE_DIR/$1.tar.gz'
+  - du -hs $DOCKER_CACHE_DIR/*
```

Comment:
While I can decipher what this does, I think having a little comment to give a clue to other maintainers could be helpful (although comments can become outdated & more dangerous than none :smile:)

Reply:
hehe... I didn't enciphered much as what this one-liner already was, but that's right that a comment will be welcome. I'll try to put something...

Reply:
@thbar comments added. Please let me know what do you think...

Reply:
@gildegoma looks good enough to me! Thanks for everyone!